src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -83,7 +84,15 @@ private CloudWatchCollector(Map<String, Object> config, AmazonCloudWatchClient c
         }
 
         if (client == null) {
-          this.client = new AmazonCloudWatchClient();
+          if (config.containsKey("role_arn")) {
+            STSAssumeRoleSessionCredentialsProvider credentialsProvider = new STSAssumeRoleSessionCredentialsProvider(
```

Comment:
Does this handle refreshing credentials as needed, or does that need to be implemented by us?